charts/kube-prometheus-stack/templates/alertmanager/alertmanager.yaml

```diff
@@ -65,7 +65,7 @@ spec:
 {{- end }}
 {{- if .Values.alertmanager.alertmanagerSpec.alertmanagerConfigNamespaceSelector }}
   alertmanagerConfigNamespaceSelector:
-{{ toYaml .Values.alertmanager.alertmanagerSpec.alertmanagerConfigNamespaceSelector | indent 4}}
+{{ tpl (toYaml .Values.alertmanager.alertmanagerSpec.alertmanagerConfigNamespaceSelector | indent 4) . }}
```

Comment:
I do not have any merge permission for kube-prometheus-stack. However, can you how me the use-case here?

Reply:
> I do not have any merge permission for kube-prometheus-stack.

darn it 😊
> However, can you how me the use-case here?

It allows you to use helm global variables like `{{.Release.Namespace}}` in values.yaml for those specific keys.

https://helm.sh/docs/howto/charts_tips_and_tricks/#using-the-tpl-function

Reply:
I can fully understand the reason, but using the `tpl (toYaml .) $` always has some limitation. For example, the pattern only works exactly to one namespace, like:

```yaml
    probeNamespaceSelector:
      matchLabels:
        kubernetes.io/name: {{ $.Release.Namespace }}
```

or

```yaml
    probeNamespaceSelector:
      matchExpressions:
        - key: name
          operator: in
          values:
            - {{ $.Release.Namespace }}
```

However, having a list of namespaces would not work here.

```mustache
    probeNamespaceSelector:
      matchExpressions:
        - key: name
          operator: in
          values: ["{{ join "\",\"" $.Values.namespaces}}"]
```

To resolve the limitation (for example on the [Alertmanager config](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/templates/alertmanager/secret.yaml#L17)), a pure string can be passed to the tpl function.

But I don't if we want this here. I would just like to share my insights with the `tpl (toYaml .) $` pattern.

Reply:
> having a list of namespaces would not work here

You can test this assumption with my pr